modules/adhashBidAdapter.js

```diff
@@ -98,17 +148,50 @@ export const spec = {
   },
 
   buildRequests: (validBidRequests, bidderRequest) => {
+    const storage = getStorageManager({ bidderCode: ADHASH_BIDDER_CODE });
     const { gdprConsent } = bidderRequest;
-    const { url } = spec;
     const bidRequests = [];
     let referrer = '';
-    if (bidderRequest && bidderRequest.refererInfo) {
-      // TODO: is 'page' the right value here?
-      referrer = bidderRequest.refererInfo.page;
+    try {
+      referrer = window.top.location.href;
+    } catch (e) {
+      referrer = window.location.href;
     }
```

Comment:
You can read the referer value from bidderRequest object, `bidderRequest.refererInfo`

Reply:
I am now using bidderRequest.refererInfo.